src/modules/m_httpd.cpp

```diff
@@ -280,11 +281,13 @@ class HttpServerSocket : public BufferedSocket, public Timer, public insp::intru
 	{
 		ModResult MOD_RESULT;
 		std::string method = http_method_str(static_cast<http_method>(parser.method));
-		HTTPRequest acl(method, uri, &headers, this, ip, body);
+		HTTPRequestURI parsed;
+		ParseURI(uri, parsed);
+		HTTPRequest acl(method, uri, parsed, &headers, this, ip, body);
 		FIRST_MOD_RESULT_CUSTOM(*aclevprov, HTTPACLEventListener, OnHTTPACLCheck, MOD_RESULT, (acl));
 		if (MOD_RESULT != MOD_RES_DENY)
 		{
-			HTTPRequest url(method, uri, &headers, this, ip, body);
+			HTTPRequest url(method, uri, parsed, &headers, this, ip, body);
```

Comment:
Is there any reason to still pass uri here?

Reply:
`HTTPRequest::GetURI()` returns the plain string prior to parsing and `GetParsedURI()` returns the post-parsing one

Reply:
Yes, but I can't see why would you ever need `GetURI()`.

Reply:
I'm not sure, the original reason was to leave it for people to use, but I guess just storing it on the parsed uri so it can be access from GetParsedURI().uri could work, idk

Reply:
You're probably better off just deleting it tbh. I can't think of a single reason why anyone would want to use it.